newIDE/app/src/UI/Theme/index.js

```diff
@@ -1,23 +1,74 @@
 // @flow
+import { createMuiTheme } from '@material-ui/core/styles';
+import { isLtr } from '../../Utils/i18n/RtlLanguages';
+import DefaultTheme, {
+  themeName as defaultThemeName,
+  type Theme,
+} from './DefaultTheme';
 import DarkTheme from './DarkTheme';
-import DefaultTheme from './DefaultTheme';
 import NordTheme from './NordTheme';
 import SolarizedDarkTheme from './SolarizedDarkTheme';
-import { type Theme } from './DefaultTheme';
 import './Global.css';
 
 // To add a new theme:
 // * copy the folder of an existing one (DarkTheme for example),
 // * import it at the top of the file
 // * add it below:
 export const themes = {
-  'GDevelop default': DefaultTheme,
+  [defaultThemeName]: DefaultTheme,
   Dark: DarkTheme,
   Nord: NordTheme,
   'Solarized Dark': SolarizedDarkTheme,
 };
 
-export const getTheme = (themeName: string): Theme =>
-  themes[themeName] || themes['GDevelop default'];
+export function getTheme({ themeName, language }: {| themeName: string, language: string |}): Theme {
```

Comment:
nitpicking:

what is returned here from this function is not a `Theme`. A `Theme` is defined to be an object like this:
```
{  gdevelopTheme, muiTheme }
```

with the content of the objects `gdevelopTheme` and `muiTheme` defined in `DefaultTheme.js` (thanks to `$Exact<typeof theme>`).

Here, what you return is an object containing `gdevelopTheme` and something else called `muiTheme`, but which is *different*, as it was passed to `createMuiTheme`. If we look at material-ui, we see that the argument is called a mui theme "options", while the returned value is a mui "theme": https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/styles/createMuiTheme.d.ts
(Unfortunately we don't use TypeScript so can't benefit from these types).

What I suggest is that:
- we keep returning `{ gdevelopTheme, muiTheme: ltr ? createLtrTheme(muiTheme) : createRtlTheme(muiTheme), };` as you're doing currently. The naming is perfectly valid!
- but we rename in DefaultTheme, and in the other theme files, `muiTheme` to `muiThemeOptions`. After all, because we removed a function call, there is no reason the object should still be called the same as before - it's something slightly different :) 
  - the only other change should be to adapt getTheme so that you pass the `muiThemeOptions` to createLtrTheme/createRtlTheme.

Finally, change the return type of getTheme. Instead of Theme, it should be something that we can call "ConstructedTheme" (or a better name?), described like this:

```
export type GDevelopTheme = $PropertyType<Theme, 'gdevelopTheme'>;
export type ConstructedTheme = {| gdevelopTheme: GDevelopTheme, muiTheme: Object |}
```

we use Object as we don't have anything better here to use.
In this approach, `Theme` is still useful, as it describe the shape of the objects used to describe the options, and it ensures consistency between themes (if you forget something in a theme, Flow will raise an error).

Let me know if something is not clear - should not be too much changes ;)